app-center-services/src/main/java/org/exoplatform/appcenter/rest/ApplicationCenterREST.java

```diff
@@ -481,6 +481,49 @@ public Response getApplicationIllustration(@Context Request request,
     }
   }
 
+  @Path(APPLICATIONS_ENDPOINT + "/search")
+  @GET
+  @Produces(MediaType.APPLICATION_JSON)
+  @RolesAllowed("users")
+  @ApiOperation(
+          value = "Search the list of applications available with query, identified by its identity technical identifier."
+              + " If no designated owner, all events available for authenticated user will be retrieved.",
+          httpMethod = "GET",
+          response = Response.class,
+          produces = "application/json"
+  )
+  @ApiResponses(
+          value = { @ApiResponse(code = HTTPStatus.OK, message = "Request fulfilled"),
+              @ApiResponse(code = HTTPStatus.UNAUTHORIZED, message = "Unauthorized operation"),
+              @ApiResponse(code = HTTPStatus.INTERNAL_ERROR, message = "Internal server error"), }
```

Comment:
There is no internal error thrown by this function
+
There is no documentation for bad request error.
You need to add them.

Reply:
ok i'll add the documentation

Reply:
Actually , for the bad request error , there is not a standard message to be returned as it depends from the offset and the limit of the search , so the messages are included in the entity of the response for each case

Reply:
Still, we have to document theose cases with the @ApiResponse as it will be used then to generate documentation with Swagger.